### DISCUSSION TITLE
Any Idea how to achieve this rather advanced attack ? (similar to traffic analysis attack)

### DISCUSSION THREAD

**Discussion by minanagehsalalma** [mitmproxy/mitmproxy#5987] [Ideas]
I made a video demonstrating the idea 

https://user-images.githubusercontent.com/20546638/224740154-36f865f8-8778-4f88-947f-469ce5bac7a2.mp4

what i am lost at is ... how to know if a user just opened an app like FB for example... so the attack become convenient

**Comment by mhils** [comment]
Cool idea, I wonder though if you can make the captive portal screen appear on a network which already provided a working connection before. :)

**Reply by minanagehsalalma** [reply]
> I wonder though if you can make the captive portal screen appear on a network which already provided a working connection before

@mhils yup this possible , my router already does that if the data plan goes out.

it can be simply done using DNS hijacking , if you didn't notice ... on android phones if the network connection goes out you can see that on the wifi icon ... it shows network status which means it keeps doing passive probing in the background.

it uses these 

> [NetworkCapabilities.NET_CAPABILITY_VALIDATED indicates that Android OS successfully validated connectivity on this network.
](https://medium.com/@veniamin.vynohradov/monitoring-internet-connection-state-in-android-da7ad915b5e5#:~:text=in%20my%20example.-,NetworkCapabilities,-.NET_CAPABILITY_INTERNET%20means%20that)

if the network connection validated check were to be redirected by a captive portal , then that would be detected too 

here is a video of me replicating it the router
i made it show the notification then made it remove it without needing to disconnect and reconnect to the network 

https://user-images.githubusercontent.com/20546638/226100661-e7f54b97-727f-4fa5-8a96-e2b0d83f44b7.mp4



on windows 

> [NCSI passive monitoring, the MSN Portal page, and the network status alert](https://learn.microsoft.com/en-us/troubleshoot/windows-client/networking/internet-explorer-edge-open-connect-corporate-public-network#ncsi-passive-monitoring-the-msn-portal-page-and-the-network-status-alert)
> In addition to the active probes that this article describes, NCSI monitors the network activity of other applications on the computer. This passive monitoring process continues even if the active probe process fails. NCSI adjusts its network status determination based on whether other applications can make successful TCP connections. If a network alert appears because of a failed active probe, it disappears when a passive probe succeeds.


[the source which is the most extensive one on the matter 
](https://textslashplain.com/2022/06/24/captive-portals/)

sorry for the late reply , hope i was able to answer your reply well.

**Reply by minanagehsalalma** [reply]
@mhils so... what do you think ?

**Reply by mhils** [reply]
Still sounds like a cool attack! I unfortunately can't help you implement it, but have fun! :)

**Reply by minanagehsalalma** [reply]
>  I unfortunately can't help you implement it
@mhils  i just need the tip of the string 😥

> what i am lost at is ... how to know if a user just opened an app like FB for example... so the attack become convenient

> but have fun! :)

thanks i will try : )
